hscontrol/app.go

```diff
@@ -489,13 +489,18 @@ func (h *Headscale) Serve() error {
 			return errSTUNAddressNotSet
 		}
 
-		region, err := h.DERPServer.GenerateRegion()
-		if err != nil {
-			return err
+		regionsCount := len(h.DERPMap.Regions)
```

Comment:
I think that we should return an error if there is a collision, as far as I understand this will cause custom DERPs to only be added when you have no other DERPs.

If the upstream or current map has the ID, we should probably log.Fatal.